feat: how-to guide for dynamic dropdown for inputs (#2130)
kestra-io/docs: content/docs/15.how-to-guides/dynamic-inputs.md

---
title: Dynamic Dropdown for Inputs
icon: /docs/icons/tutorial.svg
stage: Intermediate
topics:
  - Kestra Concepts
---

Support dynamic dropdown for inputs based on data from external source.

## Use-case

> In this guide, we will learn how can we support dynamic dropdown for inputs. The dropdown will get the values from an external source. While it is not possible to directly integrate the external source for powering the dynamic dropdown, it is possible to do so using KV store.

Reviewer: ```suggestion
In this guide, we will learn how can we support dynamic dropdown for inputs. The dropdown will get the values from an external source. While it is not possible to directly integrate the external source for powering the dynamic dropdown, it is possible to do so using the [KV store](../05.concepts/05.kv-store.md).
```

---
title: Dynamic Dropdown for Inputs
icon: /docs/icons/tutorial.svg
stage: Intermediate
topics:
  - Kestra Concepts
---

Support dynamic dropdown for inputs based on data from external source.

## Use-case

In this guide, we will learn how can we support dynamic dropdown for inputs. The dropdown will get the values from an external source. While it is not possible to directly integrate the external source for powering the dynamic dropdown, it is possible to do so using KV store.

## Update KV store on schedule

> We will create a flow that will fetch the data from external source, and set the value for a KV store key. The value will be in the form of a list of strings.

Reviewer: ```suggestion
We will create a flow that will fetch the data from external source, and set the value in the KV store. The value will be in the form of a list of strings.
```

> In this example, we will fetch the data from Postgres table. The flow will run on an hourly schedule. You can change the cron to run at different frequency on the basis of how frequently you expect the data at the source to change. If the external source is in a database that supports change data capture as in this case where we use Postgres table, you can also leverage debezium trigger and immediately have the KV store updated.

Reviewer: ```suggestion
In this example, the flow will fetch data from a Postgres table on an hourly schedule. You can change the `cron` property to run at a different frequency on the basis of how frequently you expect the data at the source to change. If the external source is in a database that supports change data capture as in this case where we use Postgres table, you can also leverage [debezium trigger](/plugins/debezium%20postgresql/triggers/io.kestra.plugin.debezium.postgres.trigger) and immediately update the KV store.
```

```yaml
id: update_kv_store
namespace: company.team

tasks:
  - id: fetch_departments
    type: io.kestra.plugin.jdbc.postgresql.Query

>     url: "jdbc:postgresql://{{ secret('POSTGRES_HOST') }}:25432/postgres"

Reviewer: ```suggestion
    url: "jdbc:postgresql://{{ secret('POSTGRES_HOST') }}:5432/postgres"
```
